lib/std/posix.zig

```diff
@@ -48,6 +48,9 @@ else switch (native_os) {
     else => struct {},
 };
 
+// riscv32-linux only has the statx syscall; no fstat or fstatat.
+const no_stat = !use_libc and native_os == .linux and builtin.cpu.arch == .riscv32;
```

Comment:
```suggestion
const no_stat = @TypeOf(system.fstatat) == void;
```

Reply:
Implication being that the syscall wrappers in `std.os.linux` should be defined as `void` if the syscall in question is not actually available on the architecture?

Is there a recommended pattern for this?